service/matching/matchingEngine.go

```diff
@@ -211,6 +216,24 @@ func (e *matchingEngineImpl) getTaskListManager(taskList *taskListID,
 	return mgr, nil
 }
 
+func (e *matchingEngineImpl) cacheTaskListByDomain(taskList *taskListID) error {
+	domainName, err := e.domainCache.GetDomainName(taskList.domainID)
+	if err != nil {
+		return err
+	}
+	cachedTL := e.taskListCache.Get(domainName)
+	var taskLists map[string]bool
+	if cachedTL != nil {
+		taskLists = cachedTL.(map[string]bool)
+	} else {
+		taskLists = map[string]bool{}
+	}
+
+	taskLists[taskList.name] = true
+	e.taskListCache.PutIfNotExist(domainName, taskLists)
```

Comment:
Looks like there might be some race condition here like two goroutines tries to cache two different tasklists at the same time, one of them will be lost. Not sure if this is an issue or not as this is just for caching. So, if not an issue, maybe add some comments explaining why the race condition is ignored.

Reply:
I think this is not an issue as we have all the tasklistCache gets populated eventually.

Reply:
Updated with a comment.